tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/ApplyCondition.cs

```diff
@@ -0,0 +1,142 @@
+﻿using System;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Text.Json;
+using System.Text.RegularExpressions;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Features;
+
+namespace Azure.Sdk.Tools.TestProxy.Common
+{
+    public class ApplyCondition
+    {
+
+        private string _uriRegex;
+        public string UriRegex { 
+            get
+            {
+                return _uriRegex;
+            }
+            set
+            {
+                Regex = new Regex(value);
+                _uriRegex = value;
+            }
+        }
+        private Regex Regex;
+
+        public ApplyCondition(string uriRegex)
```

Comment:
Do we actually need this ctor? Seems like if we are being overridden we can just use an empty ctor.

Reply:
Nevermind - I see that we don't override for each condition type - which makes sense.

Reply:
I used it in a test. You're absolutely right I should clean that out.

Reply:
Just a reminder to remove this.